### PR TITLE
fix: experimental outbound http

### PR DESCRIPTION
**Pull request by eduardomourar**
This will fix the experimental outbound HTTP support for both browser and Node.js. Additionally, the part of synckit code that matter has been brought into the `preview2-shim` package.

### CODE REVIEW

File: packages/preview2-shim/lib/http/synckit/index.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
+// Based on: https://github.com/un-ts/synckit
```
Comment (guybedford) [inline packages/preview2-shim/lib/http/synckit/index.js:1]: Do we need to include a license?

File: packages/preview2-shim/package.json [hunk 2]
```diff
@@ -5,8 +5,13 @@
   "author": "Guy Bedford, Eduardo Rodrigues<16357187+eduardomourar@users.noreply.github.com>",
   "type": "module",
   "exports": {
+    ".": {
+      "types": "./types/index.d.ts",
+      "node": "./lib/nodejs/index.js",
+      "default": "./lib/browser/index.js"
+    },
```
Comment (guybedford) [inline packages/preview2-shim/package.json:12]: How are you using this one?

File: packages/preview2-shim/package.json [hunk 3]
```diff
@@ -23,11 +28,11 @@
...
-    "url": "git+https://github.com/bytecodealliance/componentize-js.git"
+    "url": "git+https://github.com/bytecodealliance/jco.git"
   },
   "license": "(Apache-2.0 WITH LLVM-exception)",
   "bugs": {
-    "url": "https://github.com/bytecodealliance/componentize-js/issues"
+    "url": "https://github.com/bytecodealliance/jco/issues"
   },
-  "homepage": "https://github.com/bytecodealliance/componentize-js#readme"
+  "homepage": "https://github.com/bytecodealliance/jco#readme"
```
Comment (guybedford) [inline packages/preview2-shim/package.json:37]: 🙏

**Review by guybedford — APPROVED** [review]

**Comment by guybedford** [comment]
Looks good, thanks you. Just let me know about the license, then let's get this merged.

File: packages/preview2-shim/lib/http/synckit/index.js (hunk 1 above)
Comment (eduardomourar) [reply packages/preview2-shim/lib/http/synckit/index.js:1]: done

**Comment by eduardomourar** [comment]
Just to be clear, this is still following [the old definition](https://github.com/bytecodealliance/preview2-prototyping/pull/85/commits/72116e390bb6e07fdf7f14884dfd725e8a01eb3b), but I will work on getting it updated to [the new definition](https://github.com/WebAssembly/wasi-http/blob/1185b0a0224bc3e11274ad6298fd065cae472651/wit/outgoing-handler.wit) next.

**Comment by guybedford** [comment]
Yes, understood the work is in flux, and I'm fine to just have churn on this.

**Pull request merged**